File: src/cluster-service/internal/repository/postgres/sample.go
package postgres

import (
	"cluster/internal/entity"
	"cluster/pkg/psqlcli"
	"github.com/jmoiron/sqlx"
)

type SamplePostgres struct {
	db *sqlx.DB
}

func NewSamplePostgres(cli *psqlcli.Client) *SamplePostgres {
	return &SamplePostgres{db: cli.DB}
}

func (s *SamplePostgres) Get() ([]entity.Sample, error) {
	var samples []entity.Sample

	err := s.db.Select(&samples,
		`SELECT uuid, audio_path, emotion FROM sample`)
	if err != nil {
		return nil, err
	}

	return samples, nil
}

File: src/feature-service/entity/sample.py


class Sample:
    def __init__(self, uuid, audio_path, emotion, batch):
        self.uuid = uuid
        self.audio_path = audio_path
        self.emotion = emotion
        self.batch = batch

    def __dict__(self):
        return {
            "uuid": self.uuid,
            "audio_path": self.audio_path,
            "emotion": self.emotion,
            "batch": self.batch
        }
